src/middlewared/middlewared/plugins/datastore.py

```diff
@@ -145,245 +338,227 @@ def query(self, name, filters=None, options=None):
               "params": ["account.bsdusers", [ ["username", "=", "root" ] ], {"get": true}]
             }
         """
-        model = self.__get_model(name)
-        if options is None:
-            options = {}
-        else:
-            # We do not want to make changes to original options
-            # which might happen with "prefix"
-            options = options.copy()
+        table = self._get_table(name)
+
+        # We do not want to make changes to original options
+        # which might happen with "prefix"
+        options = options.copy()
```

Comment:
`options` can be `null`. We would want to ensure it's still a dict

Reply:
Addressed in fae679541e0f3e96d0d6e247eb565ba223deca69

Reply:
Did you inspect all .query usages that we dont use "null" anywhere? Nor UI? We cant change API out of the blue just because something before "didnt make sense" :P